weed/shell/command_volume_fsck.go

```diff
@@ -327,19 +349,21 @@ func (c *commandVolumeFsck) collectOneVolumeFileIds(tempFolder string, volumeId
 
 }
 
-func (c *commandVolumeFsck) collectFilerFileIds(volumeIdToServer map[uint32]VInfo, tempFolder string, writer io.Writer, verbose bool) error {
+func (c *commandVolumeFsck) collectFilerFileIds(dataNodeVolumeIdToVInfo map[string]map[uint32]VInfo, tempFolder string, writer io.Writer, verbose bool) error {
 
 	if verbose {
 		fmt.Fprintf(writer, "collecting file ids from filer ...\n")
 	}
 
 	files := make(map[uint32]*os.File)
-	for vid := range volumeIdToServer {
-		dst, openErr := os.OpenFile(getFilerFileIdFile(tempFolder, vid), os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0644)
-		if openErr != nil {
-			return fmt.Errorf("failed to create file %s: %v", getFilerFileIdFile(tempFolder, vid), openErr)
+	for _, volumeIdToServer := range dataNodeVolumeIdToVInfo {
+		for vid := range volumeIdToServer {
+			dst, openErr := os.OpenFile(getFilerFileIdFile(tempFolder, vid), os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0644)
+			if openErr != nil {
+				return fmt.Errorf("failed to create file %s: %v", getFilerFileIdFile(tempFolder, vid), openErr)
+			}
+			files[vid] = dst
```

Comment:
one replicated volume id will map to multiple dst files here.

Reply:
@chrislusf  This one set of records for the volume id from the filer will be mapped to all replicas of the volume idx files

Reply:
will this `dst` file be opened multiple times? but only closed once?

Reply:
> will this `dst` file be opened multiple times? but only closed once?

no it is checked above
https://github.com/chrislusf/seaweedfs/pull/2858/files#diff-8e0b202ea9cb980864a59d8e563827cb5748abc6ed60be322ccca4b49abb7a94R168